### PR TITLE
convert jira issues - removing backward compatible, adding links

### PR DESCRIPTION
**Pull request by kobymeir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
convert jira issues 
- removing backward compatible
- adding links to jira tickets in gitlab
- adding the host running the pipeline to the slack message
- showing playbooks and modeling errors as warning if all the tickets were update/created in jira.

### CODE REVIEW

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 1]
```diff
@@ -476,7 +476,6 @@ jobs-done-check-on-merge:
     - .push-rule
     - .jobs-done-check
   needs:
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.on-push.yml:478]:
```suggestion
  needs:
      - merge-dev-secrets
```

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 2]
```diff
@@ -476,7 +476,7 @@ jobs-done-check-on-merge:
...
-    - merge-dev-secrets
+      - merge-dev-secrets
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.on-push.yml:479]:
```suggestion
    - merge-dev-secrets
```

File: .gitlab/ci/.gitlab-ci.bucket-upload.yml [hunk 3]
```diff
@@ -730,7 +730,6 @@ sync-buckets-between-projects:
     - gke
   extends:
     - .bucket-upload-rule
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.bucket-upload.yml:732]:
```suggestion
    - .bucket-upload-rule
    - .default-job-settings
```

File: Tests/Marketplace/print_test_playbook_summary.py [hunk 4]
```diff
@@ -39,41 +41,6 @@ def read_file_contents(file_path: Path) -> list | None:
     return None
 
 
-def print_test_playbooks_summary_without_junit_report(artifacts_path: Path) -> bool:
```
Comment (kobymeir) [inline Tests/Marketplace/print_test_playbook_summary.py:42]: backward compatible code for the SDK, not needed anymore

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 5]
```diff
@@ -95,7 +95,7 @@
   - section_end "Stop contrib external build"
 
 .create_artifacts_and_server_type_instance_folders: &create_artifacts_and_server_type_instance_folders
-  - section_start "Create Artifacts & Server Instance & Server Type folders" --collapsed
+  - section_start "Create Artifacts, Server Instance, Server Type folders" --collapsed
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.global.yml:98]: The `&` causes issues in gitlab

File: Tests/scripts/gitlab_slack_notifier.py [hunk 6]
```diff
@@ -255,26 +294,35 @@ def bucket_upload_results(bucket_artifact_folder: Path,
...
+    if pull_request:
+        content_fields.append({
+            "title": "Pull Request",
+            "value": slack_link(pull_request.data['html_url'], pull_request.data['title']),
             "short": False
         })
 
     # report failing unit-tests
     triggering_workflow_lower = triggering_workflow.lower()
-    check_unittests_substrings = {'lint', 'unit', 'demisto sdk nightly', TEST_NATIVE_CANDIDATE.lower()}
```
Comment (kobymeir) [inline Tests/scripts/gitlab_slack_notifier.py:272]: moved it to be a constant

File: Tests/scripts/gitlab_slack_notifier.py [hunk 7]
```diff
@@ -309,6 +356,7 @@ def construct_slack_msg(triggering_workflow: str,
     else:
         title += ' - Success'
         color = 'good'
+    title += f' (@{CI_SERVER_HOST})' if CI_SERVER_HOST else ''
```
Comment (kobymeir) [inline Tests/scripts/gitlab_slack_notifier.py:359]: support for the XDR migration to show on which host the pipeline is running

File: Tests/scripts/test_modeling_rule_report.py [hunk 8]
```diff
@@ -121,3 +125,20 @@ def calculate_test_modeling_rule_results(test_modeling_rules_results_files: dict
                     jira_tickets_for_modeling_rule[summary] = sorted_issues_matching_summary[0]
 
     return modeling_rules_to_test_suite, jira_tickets_for_modeling_rule, server_versions
+
+
+def write_test_modeling_rule_to_jira_mapping(artifacts_path: Path, jira_tickets_for_modeling_rule: dict[str, Issue]):
+    test_modeling_rule_to_jira_mapping_file = artifacts_path / TEST_MODELING_RULES_TO_JIRA_MAPPING
+    logging.info(f"Writing test modeling rules to Jira mapping to {test_modeling_rule_to_jira_mapping_file}")
```
Comment (tkatzir) [inline Tests/scripts/test_modeling_rule_report.py:132]:
```suggestion
    logging.info(f"Writing test_modeling_rules_to_jira_mapping to {test_modeling_rule_to_jira_mapping_file}")
```

File: Tests/scripts/test_modeling_rule_report.py [hunk 9]
```diff
@@ -121,3 +125,20 @@ def calculate_test_modeling_rule_results(test_modeling_rules_results_files: dict
...
+    with open(test_modeling_rule_to_jira_mapping_file, "w") as test_modeling_rule_to_jira_mapping_fp:
+        test_modeling_rule_to_jira_mapping = {modeling_rule: jira_ticket_to_json_data(jira_ticket)
+                                              for modeling_rule, jira_ticket in jira_tickets_for_modeling_rule.items()}
+        test_modeling_rule_to_jira_mapping_fp.write(json.dumps(test_modeling_rule_to_jira_mapping, indent=4, sort_keys=True,
+                                                               default=str))
+
+
+def read_test_modeling_rule_to_jira_mapping(artifacts_path: Path) -> dict[str, dict[str, str]]:
+    with (contextlib.suppress(Exception),
```
Comment (tkatzir) [inline Tests/scripts/test_modeling_rule_report.py:141]: Or `info` You decide.
```suggestion
    logging.debug(f"Reading test_modeling_rules_to_jira_mapping from {TEST_MODELING_RULES_TO_JIRA_MAPPING}")
    with (contextlib.suppress(Exception),
```

File: Tests/scripts/gitlab_slack_notifier.py [hunk 10]
```diff
@@ -156,24 +167,45 @@ def test_modeling_rules_results(artifact_folder: Path, pipeline_url: str, title:
     }]
 
 
+def failed_test_data_to_slack_link(failed_test: str, jira_ticket_data: dict[str, str] | None) -> str:
```
Comment (tkatzir) [inline Tests/scripts/gitlab_slack_notifier.py:170]: Nice!

File: Tests/scripts/gitlab_slack_notifier.py [hunk 11]
```diff
@@ -255,26 +292,35 @@ def bucket_upload_results(bucket_artifact_folder: Path,
 
 def construct_slack_msg(triggering_workflow: str,
                         pipeline_url: str,
-                        pipeline_failed_jobs: list,
+                        pipeline_failed_jobs: list[ProjectPipelineJob],
                         pull_request: GithubPullRequest | None) -> list[dict[str, Any]]:
     # report failing jobs
     content_fields = []
-    failed_jobs_names = {job.name for job in pipeline_failed_jobs}
+    slack_msg_append = []
```
Comment (tkatzir) [inline Tests/scripts/gitlab_slack_notifier.py:299]: Why was it moved here, and not near where it's first used?

File: Tests/scripts/gitlab_slack_notifier.py [hunk 12]
```diff
@@ -360,7 +406,7 @@ def construct_coverage_slack_msg() -> list[dict[str, Any]]:
     yesterday = datetime.now() - timedelta(days=1)
     coverage_yesterday = get_total_coverage(date=yesterday)
     color = 'good' if coverage_today >= coverage_yesterday else 'danger'
-    title = f'Content code coverage: {coverage_today:.3f}%'
+    title = f'Content code coverage: {coverage_today:.3f}% (Yesterday: {coverage_yesterday:.3f}%)'
```
Comment (tkatzir) [inline Tests/scripts/gitlab_slack_notifier.py:409]: Do we have last_week and last_month?
If yes - Can you please add them here as well?

File: Tests/scripts/test_playbooks_report.py [hunk 13]
```diff
@@ -41,3 +45,21 @@ def get_jira_tickets_for_playbooks(playbook_ids: list[str],
                                    reverse=True)
             playbook_ids_to_jira_tickets[playbook_id] = sorted_issues[0]
     return playbook_ids_to_jira_tickets
+
+
+def write_test_playbook_to_jira_mapping(artifacts_path: Path, jira_tickets_for_playbooks: dict[str, Issue]):
+    test_playbooks_to_jira_mapping = artifacts_path / TEST_PLAYBOOKS_TO_JIRA_MAPPING
+    logging.info(f"Writing test playbooks to Jira mapping to {test_playbooks_to_jira_mapping}")
```
Comment (tkatzir) [inline Tests/scripts/test_playbooks_report.py:52]:
```suggestion
    logging.info(f"Writing test_playbooks_to_jira_mapping to {test_playbooks_to_jira_mapping}")
```

File: Tests/scripts/test_playbooks_report.py [hunk 14]
```diff
@@ -41,3 +45,21 @@ def get_jira_tickets_for_playbooks(playbook_ids: list[str],
...
+    with open(test_playbooks_to_jira_mapping, "w") as playbook_to_jira_mapping_file:
+        playbook_to_jira_mapping = {playbook_id: jira_ticket_to_json_data(jira_ticket)
+                                    for playbook_id, jira_ticket in jira_tickets_for_playbooks.items()}
+        playbook_to_jira_mapping_file.write(json.dumps(playbook_to_jira_mapping, indent=4, sort_keys=True,
+                                                       default=str))
+
+
+def read_test_playbook_to_jira_mapping(artifacts_path: Path):
+    with (contextlib.suppress(Exception),
```
Comment (tkatzir) [inline Tests/scripts/test_playbooks_report.py:61]:
```suggestion
    logging.debug(f"Reading test_playbooks_to_jira_mapping from {TEST_PLAYBOOKS_TO_JIRA_MAPPING}")
    with (contextlib.suppress(Exception),
```

**Review by tkatzir — APPROVED** [review]
Very good!
Approved, with some comments.

**Pull request merged**